dojo/tools/nuclei/parser.py

```diff
@@ -0,0 +1,84 @@
+import json
+import hashlib
+import hyperlink
+from dojo.models import Finding, Endpoint
+from django.core.validators import URLValidator
+from django.core.exceptions import ValidationError
+
+
+class NucleiParser(object):
+    """
+    A class that can be used to parse the nuclei JSON report file
+    """
+
+    # table to match nuclei severity to DefectDojo severity
+    SEVERITY = {
+        "critical": "Critical",
+        "high": "High",
+        "medium": "Medium",
+        "low": "Low",
+        "info": "Info",
+    }
+
+    def get_scan_types(self):
+        return ["Nuclei Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Nuclei Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON output for nuclei scan report."
+
+    def get_findings(self, filename, test):
+        data = [json.loads(line) for line in filename]
+        if len(data) == 0:
+            return []
+        else:
+            dupes = {}
+            for item in data:
+                template_id = item.get('templateID')
+                info = item.get('info')
+                name = info.get('name')
+                if info.get('severity') in self.SEVERITY:
+                    severity = self.SEVERITY[info.get('severity')]
+                else:
+                    severity = "Low"
+                type = item.get('type')
+                host = item.get('host')
+                matched = item.get('matched')
+                ip = item.get('ip')
+
+                dupe_key = hashlib.sha256(
+                    (template_id + type + host + ip + matched).encode('utf-8')
+                ).hexdigest()
+
+                if dupe_key in dupes:
+                    finding = dupes[dupe_key]
+                    finding.nb_occurences += 1
+                else:
+                    finding = Finding(
+                        title=f"{name}: {matched}",
```

Comment:
I suppose, URL shouldn't be part of the title. URL should be added as an endpoint.
Same for `dupe_key`

Reply:
I think it would be nice to leave the URL in these places, this allows you to visually and for deduplication determine different triggers, since without this defectdojo will assume that triggers for:
```
example.com/test1
example.com/test2
```
This is the same trigger.

Reply:
As I know, all findings from the dynamic analysis are every time created as one finding (if it is based on the same CVE, uses the same Title, or ...). The `Endpoints` are used to identify which on places is possible to find this vulnerability.

Reply:
Also `host` and `ip` shouldn't be in `dupe_key` (same reason as `matched`)

Reply:
I think the same. The UI already take care of endpoints. It's better to let the right data structure support this data (endpoints/URL) instead of keeping it artificially in the title.

Reply:
Fixed